components/payloads/cu_ros_payloads/src/std_msgs.rs

```diff
@@ -0,0 +1,25 @@
+use serde::{Deserialize, Serialize};
+
+use crate::RosMsgAdapter;
+
+// std_msgs/Int8 like struct
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Int8 {
+    pub data: i8,
+}
+
+impl RosMsgAdapter for i8 {
+    type Output = Int8;
+
+    fn convert(&self) -> Self::Output {
+        Int8 { data: *self }
+    }
+
+    fn namespace() -> &'static str {
+        "std_msgs"
+    }
+
+    fn type_name() -> &'static str {
```

Comment:
if the type_name is almost always the Struct name you can provide it for the user with a small default impl in the trait maybe? 

```Rust
pub trait RosMsgAdapter<'a> {
    type Output: serde::Serialize;

    fn convert(&self) -> Self::Output;

    fn namespace() -> &'a str;

    fn type_name() -> &'a str {
        ros_type_name!(Self::Output)
    }
}

#[macro_export]
macro_rules! ros_type_name {
    ($t:ty) => {{
        std::any::type_name::<$t>()
            .rsplit("::")
            .next()
            .unwrap()
    }};
}

```

link to the playground with a little test:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=c10055c62bf8aadd4e77e9920d3965a7

note that the user can override this behavior in the implementation